### ISSUE TITLE
ERR_OSSL_EVP_UNSUPPORTED running expo dev client in dev mode

### ISSUE THREAD

**Issue by justanotheratom** [aarich/charts-for-supabase#5]
Sequence of commands:
    > npm install 
    > npx expo start --dev-client
        > 'w' to run in web mode
        (Received above error)
    > set NODE_OPTIONS=--openssl-legacy-provider
        (Received above error)
    > npm audit fix --force
    > npx expo start --dev-client
        > 'w' to run in web mode
        (Bunch of source map warnings followed by:
        ERROR in ./node_modules/expo-modules-core/build/uuid/uuid.web.js:9:8
        Module not found: Can't resolve 'crypto'
           7 |   const cryptoObject =
           8 |     typeof crypto === 'undefined' || typeof crypto.randomUUID === 'undefined'
        >  9 |       ? require('crypto')
             |        ^
          10 |       : crypto;
          11 |
          12 |   if (!cryptoObject?.randomUUID) {
        
        ERROR in ./node_modules/normalize-url/index.js:2
        Module not found: Can't resolve 'url'
          1 | 'use strict';
        > 2 | const url = require('url');
          3 | const punycode = require('punycode');
          4 | const queryString = require('query-string');
          5 | const prependHttp = require('prepend-http');

Log below:

```
PS C:\Users\sanke\code\charts-for-supabase> node -v
v20.13.1
PS C:\Users\sanke\code\charts-for-supabase> npm install
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: react-native-eva-icons@1.3.1
npm warn Found: react-native-svg@13.6.0
npm warn node_modules/react-native-svg
npm warn   react-native-svg@"13.6.0" from the root project
npm warn   2 more (@ui-kitten/components, @ui-kitten/eva-icons)
npm warn
npm warn Could not resolve dependency:
npm warn peer react-native-svg@"^9.4.0" from react-native-eva-icons@1.3.1
npm warn node_modules/@ui-kitten/eva-icons/node_modules/react-native-eva-icons
npm warn   react-native-eva-icons@"^1.3.1" from @ui-kitten/eva-icons@5.1.2
npm warn   node_modules/@ui-kitten/eva-icons
npm warn
npm warn Conflicting peer dependency: react-native-svg@9.13.6
npm warn node_modules/react-native-svg
npm warn   peer react-native-svg@"^9.4.0" from react-native-eva-icons@1.3.1
npm warn   node_modules/@ui-kitten/eva-icons/node_modules/react-native-eva-icons
npm warn     react-native-eva-icons@"^1.3.1" from @ui-kitten/eva-icons@5.1.2
npm warn     node_modules/@ui-kitten/eva-icons
npm warn deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm warn deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm warn deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm warn deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm warn deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm warn deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm warn deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm warn deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm warn deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
npm warn deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm warn deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0
npm warn deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.

added 2555 packages, and audited 2556 packages in 23s

172 packages are looking for funding
  run `npm fund` for details

100 vulnerabilities (2 low, 71 moderate, 21 high, 6 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
PS C:\Users\sanke\code\charts-for-supabase> npm run dev
npm error Missing script: "dev"
npm error
npm error To see a list of scripts, run:
npm error   npm run

npm error A complete log of this run can be found in: C:\Users\sanke\AppData\Local\npm-cache\_logs\2024-05-26T23_59_50_529Z-debug-0.log
PS C:\Users\sanke\code\charts-for-supabase> npx expo start --dev-client
Starting project at C:\Users\sanke\code\charts-for-supabase
Starting Metro Bundler
▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
█ ▄▄▄▄▄ █▀ ▀ ▄▀█ ▄ ▀▄▄ ▄ ██ ▄▄▄▄▄ █
█ █   █ █▄█▄  ███ ▄▀█▄█▀  █ █   █ █
█ █▄▄▄█ █ ▀▄▄▄█▄ ▀▄▄▀▀▀▀▄▀█ █▄▄▄█ █
█▄▄▄▄▄▄▄█ ▀▄▀▄█▄▀ ▀▄▀▄▀▄█ █▄▄▄▄▄▄▄█
█ ▄█ █▀▄ ██▄██ ▀██▀▀█ ▄██ ▀ ▄▄▀▄▄▀█
██▀█   ▄█ ▄▄██ ▄█▀██ █ ▄█ █ ▄  ██▄█
█ ██ ▄█▄▀▄█▀▀ ▄ ▀▄ ▀▀▀ █▄   ▀▀▀██▀█
██▄▀▀█▀▄▄▄  ▄█ ▄█▄ ██ ▀█ ▀▀▄▄▀▄█▀██
█  █▄▄▀▄▀▀▀█▀██▄█▄▄▀█▀▄█▀ █▄▀▄██▄▄█
█ ▀ ▀██▄▄█▀ ▀▀█▄▀█▀▄▀▀▀██ █▄▀█▄█▄▄█
██ ▄█▀▀▄█▄█▄▄▄▄██▄▀▄▄█ █▄▀ ▀ █ ▀█ █
█▄ ██ ▄▄▀████▄▀▀█▀█▀ ▄ ▀█▀██ ▄▀▄█ █
█▄█▄▄▄█▄▄▀█▀██  ▀▄  ▀▀▄▀▀ ▄▄▄ ▀█▄ █
█ ▄▄▄▄▄ █▄▀▀▄█▀██▄▀ █▀▀██ █▄█ █▄▄██
█ █   █ █▀▄▀▀█▀▄█▄███▄▄▄█▄  ▄ ▄█▀▀█
█ █▄▄▄█ █▀ ▀█ ▄█▀█▀ █▀▀▄  ▀▀  ██▄▄█
█▄▄▄▄▄▄▄█▄█▄█▄▄██▄█▄██▄██▄▄█▄▄██▄██

› Metro waiting on exp+sbcharts://expo-development-client/?url=http%3A%2F%2F192.168.86.30%3A8081
› Scan the QR code above with Expo Go (Android) or the Camera app (iOS)

› Press a │ open Android
› Press w │ open web

› Press j │ open debugger
› Press r │ reload app
› Press m │ toggle menu

› Press ? │ show all commands

Logs for your project will appear below. Press Ctrl+C to exit.
Starting Webpack on port 19006 in development mode.
‼ ｢wds｣: transportMode is an experimental option, meaning its usage could potentially change without warning
▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
█ ▄▄▄▄▄ █▀ ▀ ▄▀█ ▄ ▀▄▄ ▄ ██ ▄▄▄▄▄ █
█ █   █ █▄█▄  ███ ▄▀█▄█▀  █ █   █ █
█ █▄▄▄█ █ ▀▄▄▄█▄ ▀▄▄▀▀▀▀▄▀█ █▄▄▄█ █
█▄▄▄▄▄▄▄█ ▀▄▀▄█▄▀ ▀▄▀▄▀▄█ █▄▄▄▄▄▄▄█
█ ▄█ █▀▄ ██▄██ ▀██▀▀█ ▄██ ▀ ▄▄▀▄▄▀█
██▀█   ▄█ ▄▄██ ▄█▀██ █ ▄█ █ ▄  ██▄█
█ ██ ▄█▄▀▄█▀▀ ▄ ▀▄ ▀▀▀ █▄   ▀▀▀██▀█
██▄▀▀█▀▄▄▄  ▄█ ▄█▄ ██ ▀█ ▀▀▄▄▀▄█▀██
█  █▄▄▀▄▀▀▀█▀██▄█▄▄▀█▀▄█▀ █▄▀▄██▄▄█
█ ▀ ▀██▄▄█▀ ▀▀█▄▀█▀▄▀▀▀██ █▄▀█▄█▄▄█
██ ▄█▀▀▄█▄█▄▄▄▄██▄▀▄▄█ █▄▀ ▀ █ ▀█ █
█▄ ██ ▄▄▀████▄▀▀█▀█▀ ▄ ▀█▀██ ▄▀▄█ █
█▄█▄▄▄█▄▄▀█▀██  ▀▄  ▀▀▄▀▀ ▄▄▄ ▀█▄ █
█ ▄▄▄▄▄ █▄▀▀▄█▀██▄▀ █▀▀██ █▄█ █▄▄██
█ █   █ █▀▄▀▀█▀▄█▄███▄▄▄█▄  ▄ ▄█▀▀█
█ █▄▄▄█ █▀ ▀█ ▄█▀█▀ █▀▀▄  ▀▀  ██▄▄█
█▄▄▄▄▄▄▄█▄█▄█▄▄██▄█▄██▄██▄▄█▄▄██▄██

› Metro waiting on exp+sbcharts://expo-development-client/?url=http%3A%2F%2F192.168.86.30%3A8081
› Scan the QR code above with Expo Go (Android) or the Camera app (iOS)

› Web is waiting on http://localhost:19006

› Press a │ open Android
› Press w │ open web

› Press j │ open debugger
› Press r │ reload app
› Press m │ toggle menu

› Press ? │ show all commands

› Open in the web browser...
› Press ? │ show all commands
node:internal/crypto/hash:79
  this[kHandle] = new _Hash(algorithm, xofLen, algorithmId, getHashCache());
                  ^

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:79:19)
    at Object.createHash (node:crypto:139:10)
    at module.exports (C:\Users\sanke\code\charts-for-supabase\node_modules\webpack\lib\util\createHash.js:135:53)
    at NormalModule._initBuildHash (C:\Users\sanke\code\charts-for-supabase\node_modules\webpack\lib\NormalModule.js:417:16) 
    at handleParseError (C:\Users\sanke\code\charts-for-supabase\node_modules\webpack\lib\NormalModule.js:471:10)
    at C:\Users\sanke\code\charts-for-supabase\node_modules\webpack\lib\NormalModule.js:503:5
    at C:\Users\sanke\code\charts-for-supabase\node_modules\webpack\lib\NormalModule.js:358:12
    at C:\Users\sanke\code\charts-for-supabase\node_modules\loader-runner\lib\LoaderRunner.js:373:3
    at iterateNormalLoaders (C:\Users\sanke\code\charts-for-supabase\node_modules\loader-runner\lib\LoaderRunner.js:214:10)  
    at Array.<anonymous> (C:\Users\sanke\code\charts-for-supabase\node_modules\loader-runner\lib\LoaderRunner.js:205:4)      
    at Storage.finished (C:\Users\sanke\code\charts-for-supabase\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:55:16)
    at C:\Users\sanke\code\charts-for-supabase\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:91:9
    at C:\Users\sanke\code\charts-for-supabase\node_modules\graceful-fs\graceful-fs.js:123:16
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read/context:68:3) {
  opensslErrorStack: [
    'error:03000086:digital envelope routines::initialization error',
    'error:0308010C:digital envelope routines::unsupported'
  ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

Node.js v20.13.1

```
